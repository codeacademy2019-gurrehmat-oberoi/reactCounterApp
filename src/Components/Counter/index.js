import React, { Component } from 'react';
import '../App/App.css';
import Button from '../Button';
// import '../Button';
// import Button as '../Button';
// const Button = require('../Button')

class Counter extends Component {
  state = {
    countVal : this.props.initialValue,
  }
  incrementCountVal = () => {
    this.setState({
      countVal:this.state.countVal+1
    });
  };
  decrementCountVal = () => {
    this.setState({
      countVal:this.state.countVal-1
    });
  };
  render() {
    // console.log(this);
    return (
      <div>
        <div>Counter: {this.state.countVal}</div>
        <Button caption = "PLUS" onClick = {this.incrementCountVal} />
        <Button caption = "MINUS" onClick = {this.decrementCountVal} />
      </div>
    );
  }
}

export default Counter;