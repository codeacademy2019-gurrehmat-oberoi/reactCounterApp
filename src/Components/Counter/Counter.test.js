import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../Counter';
import {shallow} from 'enzyme';

describe('Counter', () => {
  it('should render a counter', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should set state.countVal to initial value',()=>{
    const wrapper = shallow(<Counter initialValue={10}/>);
    expect(wrapper.instance().state.countVal).toEqual(wrapper.instance().props.initialValue);
  });
  it('should increment state countVal on running incrementCountVal', () => {
    const wrapper = shallow(<Counter initialValue = {10}/>);
    const origVal = wrapper.instance().state.countVal;
    wrapper.instance().incrementCountVal();
    expect(wrapper.instance().state.countVal).toEqual(origVal+1);
  });
  it('should decrement state countVal on running decrementCountVal', () => {
    const wrapper = shallow(<Counter initialValue = {10}/>);
    const origVal = wrapper.instance().state.countVal;
    wrapper.instance().decrementCountVal();
    expect(wrapper.instance().state.countVal).toEqual(origVal-1);
  })
});