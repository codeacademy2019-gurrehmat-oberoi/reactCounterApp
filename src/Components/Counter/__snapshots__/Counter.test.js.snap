// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a counter 1`] = `
<div>
  <div>
    Counter: 
  </div>
  <button
    onClick={[Function]}
    type="button"
  >
    PLUS
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    MINUS
  </button>
</div>
`;
